addons/test_mass_mailing/tests/test_mailing.py

```diff
@@ -119,6 +119,56 @@ def test_mailing_gateway_update(self):
         self.assertMailingStatistics(mailing, expected=5, delivered=4, sent=5, opened=1, clicked=1, bounced=1)
         self.assertEqual(recipients[1].message_bounce, 1)
 
+    @users('user_marketing')
+    @mute_logger('odoo.addons.mail.models.mail_mail')
+    def test_mailing_recipients(self):
+        """ Test recipient-specific computation, with email, formatting,
+        multi-emails, ... to test corner cases. """
+        customer_mult, customer_fmt = self.env['res.partner'].create([
+            {
+                'email': 'customer.multi.1@example.com, "Test Multi 2" <customer.multi.2@example.com>',
+                'name': 'MultiEMail',
+            }, {
+                'email': '"Formatted Customer" <test.customer.format@example.com>',
+                'name': 'FormattedEmail',
+            }
+        ])
+        record_p_mult, record_p_fmt, record_mult, record_fmt = self.env['mailing.test.blacklist'].create([
```

Comment:
Ok so you use "mailing.test.blacklist" because it has a customer_id field and so you can add customer-related tests?

Does not really seem related to blacklist, but there is no level between "simple" and "blacklist", maybe we should have a "mail.test.customer". Probably overkill though.

Reply:
You mad man! :smile: